src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,97 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+import java.util.Objects;
```

Comment:
Don't use Objects, Arrays, or any other util class. Instead of using Arrays.copyOf in ensureCapacity, implement your own method to resize the array.